pkg/controllers/updaterun/controller.go

```diff
@@ -31,9 +32,26 @@ import (
 	"go.goms.io/fleet/pkg/utils"
 	"go.goms.io/fleet/pkg/utils/condition"
 	"go.goms.io/fleet/pkg/utils/controller"
+	"go.goms.io/fleet/pkg/utils/controller/metrics"
 	"go.goms.io/fleet/pkg/utils/informer"
 )
 
+// updateRunMetricsStatus is the status string for update run metrics.
+type updateRunMetricsStatus string
+
+const (
+	// updateRunMetricsStatusInitializing represents the update run is initializing in the metrics.
+	updateRunMetricsStatusInitializing updateRunMetricsStatus = "initializing"
+	// updateRunMetricsStatusProgressing represents the update run is in progress in the metrics.
+	updateRunMetricsStatusProgressing updateRunMetricsStatus = "progressing"
+	// updateRunMetricsStatusStuck represents the update run is stuck waiting for a cluster to be updated in the metrics.
+	updateRunMetricsStatusStuck updateRunMetricsStatus = "stuck"
+	// updateRunMetricsStatusCompleted represents the update run is completed in the metrics.
+	updateRunMetricsStatusCompleted updateRunMetricsStatus = "completed"
+	// updateRunMetricsStatusFailed represents the update run is failed in the metrics.
+	updateRunMetricsStatusFailed updateRunMetricsStatus = "failed"
```

Comment:
it means init failed or run failed? better to distinguish them

Reply:
The failed status is used for both. I feel it not necessary to have two status as `initializing->failed` happens for init failed while `initializing->processing->failed` happens for run failed. I'm not sure if we need separate alerts for init failed and run failed.